processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -1228,4 +1229,17 @@ public Rowboat apply(Rowboat lhs, Rowboat rhs)
       return retVal;
     }
   }
+
+  static boolean isNullColumn(Iterable<String> dimValues)
+  {
+    if (dimValues == null) {
+      return true;
+    }
+    for (String val : dimValues) {
+      if (!Strings.isNullOrEmpty(val)) {
```

Comment:
I believe `val` here will always be either a non-empty string or `null`, but never empty string, since Indexed<String> returned by getDimValueLookup, which is either an Indexed<String> based on a NullValueConverterDimDim or or a GenericIndexed<String> will already map empty to null.